pkg/plugins/plugins.go

```diff
@@ -64,6 +64,8 @@ type Plugin struct {
 	client         backendplugin.Plugin
 	log            log.Logger
 
+	SkipHostEnvVars bool
```

Comment:
I decided to introduce this field here in `Plugin` (which is populated by the loader pipeline `bootstrap` step) so we don't have to repeat the check

```go
cfg.Features.IsEnabledGlobally(featuremgmt.FlagPluginsSkipHostEnvVars) &&
cfg.PluginSettings[p.ID]["skip_host_env_vars"] != "false"
```

in `envvars.go`:

https://github.com/grafana/grafana/blob/f1a830fa9f049ed1de3affa4661f79f0a67a040f/pkg/plugins/envvars/envvars.go#L91

and in the `DefaultProvider`:

https://github.com/grafana/grafana/blob/f1a830fa9f049ed1de3affa4661f79f0a67a040f/pkg/plugins/backendplugin/provider/provider.go#L69